src/components/fields/text-field/types.ts

```diff
@@ -7,7 +9,21 @@ type TCustomOptions = {
 	preventDragAndDrop?: boolean | undefined;
 };
 
+type TIconAddOn = {
+	type: "icon";
+	icon: keyof typeof Icons;
+	color?: string | undefined;
+	position?: InputGroupAddonPosition | undefined;
+};
+
+type TLabelAddOn = {
+	type: "label";
+	value: string;
+	position?: InputGroupAddonPosition | undefined;
+};
+
 export type TCustomOptionsText = TCustomOptions & {
+	addOn?: TIconAddOn | TLabelAddOn | undefined;
```

Comment:
why not add to all text fields, i.e. add to `TCustomOptions`?

Reply:
at first I didn't think label addons made sense for the email field. but actually the icon is still useful, plus the numeric field can use both too. thanks for pointing this out!